framework/src/media/utils/MediaUtils.cpp

```diff
@@ -53,25 +53,260 @@ audio_type_t getAudioTypeFromPath(std::string datapath)
 
 	toLowerString(extension);
 
-	if (extension.compare("mp3") == 0) {
+	if (extension.compare(AUDIO_TYPE_STR_MP3) == 0) {
 		medvdbg("audio type : mp3\n");
 		return AUDIO_TYPE_MP3;
-	} else if ((extension.compare("aac") == 0) || (extension.compare("mp4") == 0)) {
+	} else if ((extension.compare(AUDIO_TYPE_STR_AAC) == 0) || (extension.compare(AUDIO_TYPE_STR_MP4) == 0)) {
 		medvdbg("audio type : aac\n");
 		return AUDIO_TYPE_AAC;
-	} else if ((extension.compare("opus") == 0)) {
+	} else if ((extension.compare(AUDIO_TYPE_STR_OPUS) == 0)) {
 		medvdbg("audio type : opus\n");
 		return AUDIO_TYPE_OPUS;
-	} else if ((extension.compare("flac") == 0)) {
+	} else if ((extension.compare(AUDIO_TYPE_STR_FLAC) == 0)) {
 		medvdbg("audio type : flac\n");
 		return AUDIO_TYPE_FLAC;
-	} else if ((extension.compare("") == 0) || (extension.compare("pcm") == 0) || (extension.compare("raw") == 0)) {
+	} else if ((extension.compare(AUDIO_TYPE_STR_NULL) == 0) || (extension.compare(AUDIO_TYPE_STR_PCM) == 0) || (extension.compare(AUDIO_TYPE_STR_RAW) == 0)) {
 		medvdbg("audio type : pcm\n");
 		return AUDIO_TYPE_PCM;
+	} else if (extension.compare(AUDIO_TYPE_STR_WAV) == 0) {
+		medvdbg("audio type : wav\n");
+ 		return AUDIO_TYPE_WAVE;
 	} else {
 		medvdbg("audio type : unknown\n");
 		return AUDIO_TYPE_INVALID;
 	}
 }
+
+bool mp3_header_parsing(unsigned char *header, unsigned int *channel, unsigned int *sampleRate, audio_format_type_t *pcmFormat)
+{
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+	
+	unsigned char bit;
+	unsigned int mpegVersion;	
+	bit = header[1];
+	bit >>= 3;
+	bit &= (unsigned char)3;
+
+	switch (bit) {
+	case 0 :
+		mpegVersion = 2;
+		break;
+	case 2 :
+		mpegVersion = 1;
+		break;
+	case 3 :
+		mpegVersion = 0;
+		break;
+	default :
+		meddbg("Not Supported Format mpeg version : %u\n", mpegVersion);
+		return false;
+	}
+
+	bit = header[2];
+	bit >>= 2;
+	bit &= 3;
+	
+	switch (bit) {
+	case 0 :
+		*sampleRate = AUDIO_SAMPLE_RATE_44100;
+		break;
+	case 1 :
+		*sampleRate = AUDIO_SAMPLE_RATE_48000;
+		break;
+	case 2 :
+		*sampleRate = AUDIO_SAMPLE_RATE_32000;
+		break;
+	default :
+		meddbg("Not Supported Format sample rate : %u\n", sampleRate);
+		return false;
+	}
+
+	*sampleRate >>= mpegVersion;
+
+	bit = header[3];
+	bit >>= 6;
+	if (bit <= 2) {
+		*channel = 2;
+	} else {
+		*channel = 1;
+	}
+
+	return true;
+}
+
+bool aac_header_parsing(unsigned char *header, unsigned int *channel, unsigned int *sampleRate, audio_format_type_t *pcmFormat)
+{
+	*pcmFormat = AUDIO_FORMAT_TYPE_S16_LE;
+	
+	unsigned char bit;
```

Comment:
please add comment for each of bit operation
for example,
/** Add water for thirsty people here, Check the #of bit to check half of bottle **/